core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -223,6 +223,8 @@ object NonEmptyVector extends NonEmptyVectorInstances {
     new NonEmptyVector(buf.result)
   }
 
+  def unapply[A](nev: NonEmptyVector[A]): Some[(A, Vector[A])] = Some((nev.head, nev.tail))
```

Comment:
```scala
scala> object MyCons1 { def unapply[A](xs: scala.collection.immutable.::[A]): Some[(A, List[A])] = Some(xs.head, xs.tail) }
defined object MyCons1

scala> object MyCons2 { def unapply[A](xs: scala.collection.immutable.::[A]): Option[(A, List[A])] = Some(xs.head, xs.tail) }
defined object MyCons2

scala> def x = List(1, 2) match { case MyCons1(_, _) => }
<console>:12: warning: match may not be exhaustive.
It would fail on the following input: Nil
       def x = List(1, 2) match { case MyCons1(_, _) => }
                   ^
x: Unit

scala> def x = List(1, 2) match { case MyCons2(_, _) => }
x: Unit
```